src/SDKs/Search/DataPlane/Microsoft.Azure.Search/GeneratedSearchIndex/Models/DocumentIndexResult.cs

```diff
@@ -24,7 +24,10 @@ public partial class DocumentIndexResult
         /// <summary>
         /// Initializes a new instance of the DocumentIndexResult class.
         /// </summary>
-        public DocumentIndexResult() { }
+        public DocumentIndexResult()
+        {
+            CustomInit();
+        }
```

Comment:
Why do we have CustomInit(). Is there a model that defines it?

Reply:
@mhko This is a new feature in AutoRest. `CustomInit` is a `partial` method, so if we want we can now write custom initialization code for generated model classes. This will be very useful in the future, but we aren't yet using this capability now.

Since `CustomInit` is not defined anywhere and is marked as `partial`, the compiler will omit the call.

Reply:
Thanks Bruce

---
In reply to: [172339646](https://github.com/Azure/azure-sdk-for-net/pull/4104#discussion_r172339646) [](ancestors = 172339646)